components/CardForm/index.js

```diff
@@ -25,12 +32,34 @@ export default function CardForm({
   const { mutate } = useSWR(endpoint);
   const { data: session, status } = useSession();
   const { setIsEditMode, isEditMode } = useStore();
+  const [selectedDate, setSelectedDate] = useState(null);
+  const [showDayPicker, setShowDayPicker] = useState(false);
+  const [showTimePicker, setShowTimePicker] = useState(false);
+
+  const toggleTimePicker = () => {
+    setShowTimePicker(!showTimePicker);
+  };
+
+  const inputRef = useRef(null);
+
+  const handleDayClick = (day) => {
+    setSelectedDate(day);
+    setShowDayPicker(false);
+  };
+
+  const handleInputChange = (event) => {
+    const inputValue = event.target.value;
+    setSelectedDate(inputValue ? new Date(inputValue) : null);
+  };
+
+  const toggleDayPicker = () => {
+    setShowDayPicker((prev) => !prev);
+  };
```

Comment:
Not necessary, but we could use an implicit return if this is all we are going to do here. 

```suggestion
  const toggleDayPicker = () => setShowDayPicker((prev) => !prev);

```